docs/docs/how-tos/tool-calling.md

```diff
@@ -0,0 +1,1008 @@
+
+```python
+from langchain_core.tools import tool
+
+# highlight-next-line
+@tool("multiply_tool")
+def multiply(a: int, b: int) -> int:
+    """Multiply two numbers."""
+    return a * b
+```
+
+## 3. Context Management
+
+Tools within LangGraph sometimes require context data, such as runtime-only arguments (e.g., user IDs or session details), that should not be controlled by the model. LangGraph provides three methods for managing such context:
+
+| Type                                    | Usage Scenario                           | Mutable | Lifetime                 |
+|-----------------------------------------|------------------------------------------|---------|--------------------------|
+| [Configuration](#configuration)         | Static, immutable runtime data           | ❌       | Single invocation        |
+| [Short-term memory](#short-term-memory) | Dynamic, changing data during invocation | ✅       | Single invocation        |
+| [Long-term memory](#long-term-memory)   | Persistent, cross-session data           | ✅       | Across multiple sessions | 
+
+### Configuration
+
+Use configuration when you have **immutable** runtime data that tools require, such as user identifiers. You pass these arguments via [`RunnableConfig`](https://python.langchain.com/docs/concepts/runnables/#runnableconfig) at invocation and access them in the tool:
+
+```python
+from langchain_core.tools import tool
+from langchain_core.runnables import RunnableConfig
+
+@tool
+# highlight-next-line
+def get_user_info(config: RunnableConfig) -> str:
+    """Retrieve user information based on user ID."""
+    user_id = config["configurable"].get("user_id")
+    return "User is John Smith" if user_id == "user_123" else "Unknown user"
+
+# Invocation example with an agent
+agent.invoke(
+    {"messages": [{"role": "user", "content": "look up user info"}]},
+    # highlight-next-line
+    config={"configurable": {"user_id": "user_123"}}
+)
+```
+
+??? example "Extended example: Access config in tools"
+
+    ```python
+    from langchain_core.runnables import RunnableConfig
+    from langchain_core.tools import tool
+    from langgraph.prebuilt import create_react_agent
+    
+    def get_user_info(
+        # highlight-next-line
+        config: RunnableConfig,
+    ) -> str:
+        """Look up user info."""
+        # highlight-next-line
+        user_id = config["configurable"].get("user_id")
+        return "User is John Smith" if user_id == "user_123" else "Unknown user"
+    
+    agent = create_react_agent(
+        model="anthropic:claude-3-7-sonnet-latest",
+        tools=[get_user_info],
+    )
+    
+    agent.invoke(
+        {"messages": [{"role": "user", "content": "look up user information"}]},
+        # highlight-next-line
+        config={"configurable": {"user_id": "user_123"}}
+    )
+    ```
+
+### Short-term memory
+
+Short-term memory maintains **dynamic** state that changes during a single execution. 
```

Comment:
Link to memory

Reply:
Might make sense to also get rid of the tools sections in the memory guide

Reply:
Yes -- i added to list of outstanding items on notion so we don't forget